src/app/shared/dropdown/dropdown.directive.ts

```diff
@@ -0,0 +1,23 @@
+import { Directive, HostListener, HostBinding, ElementRef } from '@angular/core';
```

Comment:
We don't need a dropdown directive. There is one available from ng-bootstrap, https://ng-bootstrap.github.io/#/components/dropdown/examples.